### PR TITLE
Add more details to README

### PR DESCRIPTION
**Pull request by poornachandra**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+
+The Cask Data Application Platform (CDAP) Datasets provide an abstraction to store data. In this guide you will learn
+how to access data in a Dataset from a BI (Business Intelligence) Tool, allowing for ad-hoc exploration of the Dataset.
+
+What You Will Build
+-------------------
+* You will build a CDAP `Application <http://docs.cdap.io/cdap/current/en/dev-guide.html#applications>`_ that consumes
+ purchase events from a `Stream <http://docs.cdap.io/cdap/current/en/dev-guide.html#streams>`_ and store it into a
+ `Dataset <http://docs.cdap.io/cdap/current/en/dev-guide.html#datasets>`_, which is then accessed from the BI Tool.
+* You’ll build a `Flowlet <http://docs.cdap.io/cdap/current/en/dev-guide.html#flowlets>`_ that processes purchase
```
Comment (anwar6953) [inline README.rst:12]: purchase -> purchases

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+Following sections will guide you through building an application from scratch.
+If you are interested in deploying and running the application right away, you
+can clone its source code and binaries from this github repository. In that case feel
+free to skip the next two sections and jump right to Build & Run section.
+
+Application Design
+~~~~~~~~~~~~~~~~~~
+In this example we will be building a Purchase Tracker application to explore purchase events. A purchase event
+contains -
+* Customer
```
Comment (anwar6953) [inline README.rst:41]: Preview this in GitHub. Its formatting is unusual.

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+
+  export PATH=$PATH:<CDAP home>/bin
+
+We can then deploy the application to a standalone CDAP installation::
+
+  cdap-cli.sh deploy app target/cdap-bi-guide-1.0.0.jar
+  cdap-cli.sh start flow PurchaseApp.PurchaseFlow
+
+Next, we will send some sample purchase events into the stream for processing::
+  cdap-cli.sh send stream purchases "Tom,    5,       pear"
```
Comment (anwar6953) [inline README.rst:260]: Need better formatting here. In Github, it looks unusual.

File: README.rst [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+Extend This Example
+-------------------
+
+You can ask more questions like -
+  * How much sales did a product do in a day?
+  * What are the top three most popular products?
+
+If you add zip code to the purchase event, then you can ask location based questions like -
+  * What are the popular products in any zip?
+  * Which locations have most revenue?
```
Comment (anwar6953) [inline README.rst]: change locations -> regions.

locations sounds too discrete for this usage. What do you think?

File: README.rst [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+
+Share & Discuss
+---------------
+
```
Comment (anwar6953) [inline README.rst:295]:
```
Share & Discuss!
----------------
Have a question? Discuss at `CDAP User Mailing List <https://groups.google.com/forum/#!forum/cdap-user>`_

.. |(AppDesign)| image:: docs/img/app-design.png
```

File: README.rst (hunk 1 above)
Comment (poornachandra) [inline README.rst:12]: purchase events

**Pull request merged**